sdks/java/io/bigquery-io-perf-tests/src/test/java/org/apache/beam/sdk/bigqueryioperftests/BigQueryIOIT.java

```diff
@@ -111,34 +128,38 @@ public static void setup() throws IOException {
     tempRoot = options.getTempRoot();
     sourceOptions =
         SyntheticOptions.fromJsonString(options.getSourceOptions(), SyntheticSourceOptions.class);
-    bigQueryMetricsDataset = options.getBigQueryDataset();
-    bigQueryMetricsTable = options.getBigQueryTable();
-    bigQuerySourceDataset = options.getBigQuerySourceDataset();
-    bigQuerySourceTable = options.getBigQuerySourceTable();
+    metricsBigQueryDataset = options.getMetricsBigQueryDataset();
+    metricsBigQueryTable = options.getMetricsBigQueryTable();
+    testBigQueryDataset = options.getTestBigQueryDataset();
+    testBigQueryTable = options.getTestBigQueryTable();
+    writeMethod = options.getWriteMethod();
     BigQueryOptions bigQueryOptions = BigQueryOptions.newBuilder().build();
     tableQualifier =
         String.format(
-            "%s:%s.%s", bigQueryOptions.getProjectId(), bigQuerySourceDataset, bigQuerySourceTable);
-    if (!checkForSourceTable()) {
-      createAndFillSourceTable();
-    }
+            "%s:%s.%s", bigQueryOptions.getProjectId(), testBigQueryDataset, testBigQueryTable);
   }
 
   @AfterClass
   public static void tearDown() {
     BigQueryOptions options = BigQueryOptions.newBuilder().build();
     BigQuery client = options.getService();
-    TableId tableId =
-        TableId.of(options.getProjectId(), bigQuerySourceDataset, bigQuerySourceTable);
+    TableId tableId = TableId.of(options.getProjectId(), testBigQueryDataset, testBigQueryTable);
     client.delete(tableId);
   }
 
-  private static void createAndFillSourceTable() {
+  @Test
+  public void testWriteThenRead() {
+    testWrite();
+    testRead();
+  }
 
-    Pipeline pipeline = Pipeline.create();
+  private void testWrite() {
+    Pipeline pipeline = Pipeline.create(options);
 
+    BigQueryIO.Write.Method method = BigQueryIO.Write.Method.valueOf(writeMethod);
```

Comment:
You can make this one variable less - no need to hold string write method, just do this in setup: 

``` writeMethod = BigQueryIO.Write.Method.valueOf(options.getWriteMethod);```